Modules/Messages/src/Messages.Application/Infrastructure/Persistence/Repository/IMessagesRepository.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Backbone.Modules.Messages.Application.Messages.Commands.SendMessage;
+using Backbone.Modules.Messages.Application.Messages.Queries.ListMessages;
+using Backbone.Modules.Messages.Domain.Entities;
+using Backbone.Modules.Messages.Domain.Ids;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.Database;
+using Enmeshed.BuildingBlocks.Application.Pagination;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace Backbone.Modules.Messages.Application.Infrastructure.Persistence.Repository;
+public interface IMessagesRepository
+{
+    Task<DbPaginationResult<Message>> FindAll(PaginationFilter paginationFilter);
+    Task<DbPaginationResult<Message>> FindMessagesOfIdentity(IdentityAddress identityAddress, ListMessagesQuery request, IdentityAddress addressOfActiveIdentity, CancellationToken cancellationToken);
+    Task<Message> Find(MessageId id, IdentityAddress address, CancellationToken cancellationToken);
+    Task<MessageId> Add(Message message, CancellationToken cancellationToken);
+    Task<int> CountUnreceivedMessagesFromActiveIdentity(IdentityAddress sender, SendMessageCommandRecipientInformation recipientDto, CancellationToken cancellationToken);
+    EntityEntry<Message> Update(Message message);
```

Comment:
`EntityEntry` is a EF Core class. Having this in `Messages.Application` disables us to remove the reference to EF Core from this project. Just return `Task`.